### PR TITLE
pf.conf.5: revise divert-to and divert-reply

### PR DESCRIPTION
**Pull request by ihoro**
Revise `man 5 pf.conf` after recent work around pf divert(4) support.

Result preview:
283831500-8c205e64-02f8-4361-a666-ada78cc2297d

### CODE REVIEW

**Comment by ihoro** [comment]
Preliminarily discussed with @kprovost.

File: share/man/man5/pf.conf.5 [hunk 1]
```diff
@@ -2151,19 +2151,19 @@ Only effective before the route lookup happened, i.e. when filtering inbound.
...
-.Ar host
-and
+Used to
+.Xr divert 4
+packets to the given divert
 .Ar port .
-The packets will not be modified, so
-.Xr getsockname 2
-on the socket will return the original destination address of the packet.
+Historically OpenBSD pf has another meaning for this, and FreeBSD pf uses
```
Comment (jlduran) [inline share/man/man5/pf.conf.5:2158]: OpenBSD -> .Ox
FreeBSD -> .Fx

File: share/man/man5/pf.conf.5 [hunk 2]
```diff
@@ -2151,19 +2151,19 @@ Only effective before the route lookup happened, i.e. when filtering inbound.
...
+this syntax to support
+.Xr divert 4 instead. Hence,
+.Ar host
+has no meaning and can be set to anything like 127.0.0.1.
+If a packet is re-injected and does not change direction then it won't be
```
Comment (jlduran) [inline share/man/man5/pf.conf.5:2163]: igor complains about a contraction here (won't -> will not), not necessarily a requirement.

File: share/man/man5/pf.conf.5 [hunk 3]
```diff
@@ -2151,19 +2151,19 @@ Only effective before the route lookup happened, i.e. when filtering inbound.
...
+re-diverted.
 .It Ar divert-reply
-Used to receive replies for sockets that are bound to addresses
-which are not local to the machine.
-See
-.Xr setsockopt 2
-for information on how to bind these sockets.
+It has no meaning in FreeBSD pf.
```
Comment (jlduran) [inline share/man/man5/pf.conf.5:2166]: FreeBSD -> .Fx

**Review by jlduran — COMMENTED** [review]
Just a few minor nits.

**Comment by ihoro** [comment]
Thank you for the review. I've completely forgot about [testing](https://docs.freebsd.org/en/books/fdp-primer/manual-pages/#manual-pages-testing).

**Comment by ihoro** [comment]
Committed.

**Pull request closed without merging**